.github/scripts/test_agent_check.rb

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+
+require "net/http"
+require "uri"
+require "json"
+
+# Disable exiting on command failure
+agent_host = ENV["DD_AGENT_HOST"] || raise("DD_AGENT_HOST is not set")
+agent_port = ENV["DD_TRACE_AGENT_PORT"] || raise("DD_TRACE_AGENT_PORT is not set")
+
+begin
+  # Check if test agent is running
+  summary_uri = URI.parse("http://#{agent_host}:#{agent_port}/test/trace_check/summary")
+  summary_response = Net::HTTP.get_response(summary_uri)
+
+  if summary_response.code == "200"
+    puts "APM Test Agent is running. (HTTP 200)"
+  else
+    puts "APM Test Agent is not running and was not used for testing. No checks failed."
+    exit 0
+  end
+
+  # Check for test failures
+  failures_uri = URI.parse("http://#{agent_host}:#{agent_port}/test/trace_check/failures")
+  failures_response = Net::HTTP.get_response(failures_uri)
```

Comment:
Just to be sure, do we want to put some timeout settings on the request? Just in case?

Reply:
The change is actually migrate from `.circleci` with extracting environment variables. 

This has been running for quite some time, so I guess it is work quite well. Of course, we can always revisit this if we  encounter some issue.